src/lib/Preview.svelte

```diff
@@ -3,36 +3,43 @@
 
   export type CSS = Record<string, string>;
 
-  type ScenarioSize = { width?: string; height?: string; children?: number };
-  export type Scenario<C extends SvelteComponent = SvelteComponent> = {
+  type ScenarioSize = { width?: string; height?: string };
+  export type Scenario<
+    C extends SvelteComponent = SvelteComponent,
+    S = boolean,
+  > = {
     props: ComponentProps<C>;
+    slotData?: S;
     css?: CSS;
     size?: ScenarioSize;
   };
-  export type Scenarios<C extends SvelteComponent> = {
-    [key: string]: Scenario<C>;
+  export type Scenarios<C extends SvelteComponent, S = boolean> = {
+    [key: string]: Scenario<C, S>;
   };
 
   export type ScenarioState = Required<Scenario> & {
     size: Required<ScenarioSize>;
   };
 </script>
 
-<script lang="ts" generics="_C extends SvelteComponent">
+<script lang="ts" generics="_C extends SvelteComponent, _S">
   import { onMount, type ComponentType, type ComponentEvents } from "svelte";
   import ScenarioView from "$lib/ScenarioView.svelte";
 
   // _C is defined in the `generics` attribute of the `script` tag
   // but this is not recognized by eslint
   type C = _C; // eslint-disable-line no-undef
+  type S = _S; // eslint-disable-line no-undef
 
   export let component: ComponentType<C>;
-  export let scenarios: Scenarios<C>;
-  export let defaultCss: CSS = {};
+  export let scenarios: Scenarios<C, S>;
   export let emits: Extract<keyof ComponentEvents<C>, string>[] = [];
   export let setTitle: boolean = true;
   export let columns = Math.ceil(Math.sqrt(Object.keys(scenarios).length));
 
+  // create copy as array which is iterable AND modifyable
+  let scenarioArray = Object.entries(scenarios);
```

Comment:
is this variable only set here? Than it can be `const`

Reply:
I declared  `scenarioArray` with `let` as it had to be modifyable. In svelte reactive arrays can only be modified by replacing the whole array (requiring `let` instead of `const`).

But interestingly the binding in the `{#each}' loops seems not to require the `let` => I cannot why `const` works but it works. It looks to me as `{#each}` does some internal magic that works around the `const`. In this case I propose to stay with `let` to clarify that this array is modifyable (its more explicit).

If you have an explanation why `const` is correct then lets go with `const`.

Reply:
The re-assignment in svelte for reactive arrays is only needed because the compiler needs a hint, that the array has changed.

Maybe some magic in the svelte compiler gets the same hint from `{#each}`

I would use `const`, because `let` is only a workaround for the svelte 4 compiler (svelte 5 does not need the re-assignment anymore), but if you would like to keep `let` is also ok for me.

Reply:
now I switched to `const` but also renamed the variable to `editableArray`. Maybe this clarifies a little bit why this variable exists at all (see your [first question](https://github.com/stefanhoelzl/preview.svelte/pull/4#discussion_r1482251791) about it)